### PR TITLE
ROX-12113: Adding a boolean config to the email notifier, to allow unauthenticated SMTP and relevant changes to notifier to process the config element.

### PR DESCRIPTION
**Pull request by clickboo**
## Description
* New boolean in email notifier proto message to allow unauthenticated smtp. False by default.
* Skip sending the AUTH message when unauthenticated smtp is opted for
* 
## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~
- [x] Evaluated and added CHANGELOG entry if required
- [x] ~Determined and documented upgrade steps~
- [x] ~Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed
Manual testing using postfix and maildev

### CODE REVIEW

**Comment by pedrottimark** [comment]
@clickboo The change to notifier.proto file looks good from frontend viewpoint.

However, this pull request references the entire epic. How about a task instead?

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Entries in this file should be limited to:
 Please avoid adding duplicate information across this changelog and JIRA/doc input pages.
 
 ## [NEXT RELEASE]
+- ROX-11348: The email notifier now allows for unauthenticated smtp. By default, unauthenticated smtp is turned off.
```
Comment (vjwilson) [inline CHANGELOG.md:8]: "unauthenticate" and "turned off" is a little confusing, a sort of double-negative.

```suggestion
- ROX-11348: The email notifier now allows for unauthenticated SMTP. By default, authentication is still required for an email notifier, but the user can now turn it off.
```

File: generated/storage/notifier.pb.go [hunk 2]
```diff
@@ -3006,6 +3026,9 @@ func (m *Email) Size() (n int) {
 	if m.StartTLSAuthMethod != 0 {
 		n += 1 + sovNotifier(uint64(m.StartTLSAuthMethod))
 	}
+	if m.AllowUnauthenticatedSmtp {
+		n += 2
+	}
```
Comment (vjwilson) [inline generated/storage/notifier.pb.go:3031]: Whatever this is, it looks frighteningly brittle.

**Review by vjwilson — COMMENTED** [review]
Looks as sane as any Go code I have seen

File: generated/storage/notifier.pb.go (hunk 2 above)
Comment (clickboo) [reply generated/storage/notifier.pb.go:3031]: Wait what, I've never noticed this in generated code. Crazy. But protobuf.

**Comment by clickboo** [comment]
> Looks as sane as any Go code I have seen

Not saying much there, but I'll take it :-)

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,9 @@ Entries in this file should be limited to:
 Please avoid adding duplicate information across this changelog and JIRA/doc input pages.
 
 ## [NEXT RELEASE]
+- ROX-11348: The email notifier now allows for unauthenticated smtp.
+  The email notifier now allows for unauthenticated SMTP. By default,
```
Comment (theencee) [inline CHANGELOG.md:9]: "The email notifier now allows for unauthenticated SMTP" is repeated twice. Also put both sentences into one line?

File: proto/storage/notifier.proto [hunk 4]
```diff
@@ -82,6 +82,7 @@ message Email {
         LOGIN    = 2;
     }
     AuthMethod startTLSAuthMethod = 8;
+    bool allow_unauthenticated_smtp = 9;
```
Comment (theencee) [inline proto/storage/notifier.proto:85]: Please add a comment above the line. It'll automatically show up in the docs

File: central/notifiers/email/email.go [hunk 5]
```diff
@@ -128,11 +128,14 @@ func validate(email *storage.Email) error {
 	if email.GetSender() == "" {
 		errorList.AddString("Sender must be specified")
 	}
-	if email.GetUsername() == "" {
-		errorList.AddString("Username must be specified")
-	}
-	if email.GetPassword() == "" {
-		errorList.AddString("Password must be specified")
+	// username and password are optional for unauthenticated smtp
```
Comment (theencee) [inline central/notifiers/email/email.go:131]: Please add a unit test for this

File: central/notifiers/email/email.go [hunk 6]
```diff
@@ -392,8 +395,10 @@ func (e *email) send(ctx context.Context, m *message) error {
 		}
 	}
 
-	if err = client.Auth(auth); err != nil {
-		return createError("SMTP authentication failed", err)
+	if !e.notifier.GetEmail().GetAllowUnauthenticatedSmtp() {
```
Comment (theencee) [inline central/notifiers/email/email.go:398]: I'm not sure if the existing unit tests are set up for it, but it would be good to get a unit test for this

**Review by theencee — COMMENTED** [review]
lgtm but please add a unit test and fix changelog. Also do you mind creating a ticket for the integration tests within the epic? Make it blocked by [this ticket](https://issues.redhat.com/browse/ROX-8113)

File: central/notifiers/email/email.go (hunk 6 above)
Comment (clickboo) [reply central/notifiers/email/email.go:398]: Added overall test unauth email in the existing integration test

**Comment by clickboo** [comment]
> lgtm but please add a unit test and fix changelog. Also do you mind creating a ticket for the integration tests within the epic? Make it blocked by [this ticket](https://issues.redhat.com/browse/ROX-8113)

will create ticket, but the current integration tests are being skipped so the new one I added is also failing possibly because the email infra is kaput which you are taking on with ROX-8113

**Comment by clickboo** [comment]
> > lgtm but please add a unit test and fix changelog. Also do you mind creating a ticket for the integration tests within the epic? Make it blocked by [this ticket](https://issues.redhat.com/browse/ROX-8113)
> 
> will create ticket, but the current integration tests are being skipped so the new one I added is also failing possibly because the email infra is kaput which you are taking on with [ROX-8113](https://issues.redhat.com/browse/ROX-8113)

https://issues.redhat.com/browse/ROX-12114

**Review by vjwilson — APPROVED** [review]

**Pull request merged**